### PR TITLE
[qwik-pod] 29 Issue pull request dynamic card

### PR DESCRIPTION
**Pull request by AllStackDev1**
Close: https://github.com/thisdot/starter.dev-github-showcases/issues/849

## Acceptance

- [x] Card will show:
    - [x] proper icon
    - [x] title of issue / PR
    - [x] “{issue / PR number} by {author} was {open / closed} {created date}”
    - [x] comment icon
    - [x] count of comments

## Opened Issue
image

## Resolved Issue
image

## Close Unresolved Issue
image

## Opened PR
image

## Draft PR
image

## Mered PR
image

## Closed PR
image

### CODE REVIEW

**Comment by hdJerry** [comment]
@AllStackDev1 i would like to know if you considered all the possible states on this card? 
- closed
- merged
- open

File: qwik-graphql-tailwind/src/components/issue-pr-card/card.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          </span>
+        </div>
+      </div>
+
+      <div className={styles.card_right_side}>
+        <span className="ml-2 flex-1 flex-shrink-0">
+          {data.commentsCount > 0 && (
+            <a href="#" className="">
+              <div className="flex items-center justify-end">
+                <svg
```
Comment (hdJerry) [inline qwik-graphql-tailwind/src/components/issue-pr-card/card.tsx:51]: can we take this to the icon folder for usability?

File: qwik-graphql-tailwind/src/components/issue-pr-card/card.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+export const IssuePrCard = component$(({ data, type }: IssuePrCardProps) => {
+  return (
+    <div className={styles.card_container}>
+      <div className="flex">
+        <label className={styles.card_checkbox}>
+          <input type="checkbox" name="issues[]" autoComplete="off" />
+        </label>
+
+        <div className="flex-shrink-0 pl-4">
+          <span>{type == 'issue' ? <IssuesIcon className="w-5 h-5" /> : <PullRequestIcon className="w-5 h-5" />}</span>
```
Comment (hdJerry) [inline qwik-graphql-tailwind/src/components/issue-pr-card/card.tsx:28]: this icons varies, please check. there are different icons closed PR/ISSue and the colors for every state are different

**Review by hdJerry — CHANGES_REQUESTED** [review]

**Comment by AllStackDev1** [comment]
> @AllStackDev1 i would like to know if you considered all the possible states on this card?
> 
> * closed
> * merged
> * open

hmm the acceptance was for open and close, but I do understand we have another state for merged. will update that.

File: qwik-graphql-tailwind/src/components/issue-pr-card/card.tsx (hunk 2 above)
Comment (AllStackDev1) [reply qwik-graphql-tailwind/src/components/issue-pr-card/card.tsx:28]: Yes when the issue or PR is closed the icon changes, will add that also.

File: qwik-graphql-tailwind/src/components/issue-pr-card/card.tsx (hunk 1 above)
Comment (AllStackDev1) [reply qwik-graphql-tailwind/src/components/issue-pr-card/card.tsx:51]: Sure

**Comment by hdJerry** [comment]
> > @AllStackDev1 i would like to know if you considered all the possible states on this card?
> > 
> > * closed
> > * merged
> > * open
> 
> It is either opened or closed, merged is under closed.

Merged and closed are not the same. colors and icon also varies

**Comment by AllStackDev1** [comment]
> > > @AllStackDev1 i would like to know if you considered all the possible states on this card?
> > > 
> > > * closed
> > > * merged
> > > * open
> > 
> > 
> > It is either opened or closed, merged is under closed.
> 
> Merged and closed are not the same. colors and icon also varies

Yeah true, would revert to using status instead of boolean

**Comment by hdJerry** [comment]
> > > > @AllStackDev1 i would like to know if you considered all the possible states on this card?
> > > > 
> > > > * closed
> > > > * merged
> > > > * open
> > > 
> > > 
> > > It is either opened or closed, merged is under closed.
> > 
> > 
> > Merged and closed are not the same. colors and icon also varies
> 
> Yeah true, hmm, do you have idea how the status is been tracked

from the query. closed pr or issue has two statuses merged or closed. While open has just status of open

File: qwik-graphql-tailwind/src/components/icons/merged-pr.icon.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { component$, useStylesScoped$ } from '@builder.io/qwik';
+import { iconStyles } from './icons.styles';
+import { IconProps } from './types';
+
+export const MergedPrIcon = component$(({ className }: IconProps) => {
```
Comment (hdJerry) [inline qwik-graphql-tailwind/src/components/icons/merged-pr.icon.tsx:5]: mergedPricon is just PullrequestIcon with color of purple and we already have pullrequest icon

File: qwik-graphql-tailwind/src/components/issue-pr-card/card.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+// An issue can be open, closed, or resolved
+// A PR can be open, closed, or merged
+// A draft PR can be open or closed
+// when isDraft is true, the pr is open, only pr can be draft
+// when isResolved is true, the issue is closed, only issue can be resolved
+// when isMerged is true, the pr is closed, only pr can be merged
+export const IssuePrCard = component$(({ data, type }: IssuePrCardProps) => {
+  const iconColor = data.isOpen
```
Comment (hdJerry) [inline qwik-graphql-tailwind/src/components/issue-pr-card/card.tsx:38]: please, consider using classnames.

**Review by hdJerry — CHANGES_REQUESTED** [review]
please, I drop some comments above aside  that is seems good to go.

**Review by hdJerry — APPROVED** [review]

**Review by hdJerry — APPROVED** [review]

File: qwik-graphql-tailwind/src/components/issue-pr-card/card.tsx (hunk 4 above)
Comment (AllStackDev1) [reply qwik-graphql-tailwind/src/components/issue-pr-card/card.tsx]: Done

**Pull request merged**